holoviews/plotting/bokeh/chart.py

```diff
@@ -931,7 +931,7 @@ def get_data(self, element, ranges, style):
                 if is_dt:
                     width *= xdiff.astype('timedelta64[ms]').astype(np.int64)
                 else:
-                    width /= xdiff
+                    width = width * xdiff if np.min(xdiff) < 1 else width / xdiff
```

Comment:
Can you check if this is done correctly on the other plotting backends? 

(If you haven't already)

Reply:
Yes I checked matplotlib
<img width="903" alt="image" src="https://github.com/user-attachments/assets/1b48c356-3466-4e0c-a77e-6061c5f7ef98">
<img width="404" alt="image" src="https://github.com/user-attachments/assets/bfed28bb-c39d-4dfe-b876-f7f2bd8e78d9">